src/savegame_json.cpp

```diff
@@ -567,12 +567,7 @@ void Character::load( const JsonObject &data )
 
     std::string tmove_mode;
     data.read( "move_mode", tmove_mode );
-    for( int i = 0; i < CMM_COUNT; ++i ) {
-        if( tmove_mode == character_movemode_str[i] ) {
-            move_mode = static_cast<character_movemode>( i );
-            break;
-        }
-    }
+    move_mode = io::string_to_enum<character_movemode>( tmove_mode );
```

Comment:
The JSON reading code has the enum string conversion built-in; this shouldn't need the intermediate string variable; you can just read the enum directly.